server/service/live_queries.go

```diff
@@ -270,20 +270,16 @@ func (svc *Service) RunLiveQueryDeadline(
 			}
 			queryID = campaign.QueryID
 
-			readChan, cancelFunc, err := svc.GetCampaignReader(ctx, campaign)
-			if err != nil {
-				resultsCh <- fleet.QueryCampaignResult{QueryID: queryID, Error: ptr.String(err.Error()), Err: err}
-				return
-			}
-			defer cancelFunc()
-
+			// We do not want to use the outer `ctx` directly because we want to cleanup the campaign
+			// even if the outer `ctx` is canceled (e.g. a client terminating the connection).
+			// Also, we make sure stats and activity DB operations don't get killed after we return results.
+			ctxWithoutCancel := context.WithoutCancel(ctx)
```

Comment:
Curious as to why this needs to be outside.

The fix is moving the GetCampaignReader down right?

Reply:
Maybe it doesn't need to be. I put it outside defer when I was using it.

Yes, moving the GetCampaignReader down may be the actual fix. Not known/confirmed yet.